aten/src/ATen/native/TensorShape.cpp

```diff
@@ -2175,7 +2175,14 @@ std::vector<Tensor> meshgrid(TensorList tensors) {
 // Numpy-style `a.T`: returns the tensor
 // with dims reversed
 Tensor numpy_T(const Tensor &self) {
-  int64_t n = self.dim();
+  const auto n = self.dim();
+  if (n != 2 && n != 0) {
+    TORCH_WARN_ONCE(
+        "The use of `x.T` on tensors of dimension other than 2 to reverse their shape is deprecated ",
```

Comment:
It's unfortunate we don't have a good consistent way of referring to the tensor whose attribute is accessed. Maybe we can just drop the `x` and say "The use of .T on tensors..."?

Reply:
I prefer the user of `x` as a placeholder, same as in
https://github.com/pytorch/pytorch/blob/dbfd4d0d563a08a349a8105786d0aa3ee2c09823/torch/_tensor_docs.py#L4845

Reply:
Sure